src/adapter/src/util.rs

```diff
@@ -189,13 +189,13 @@ pub fn index_sql(
     use mz_sql::ast::{Expr, Value};
 
     CreateIndexStatement::<Raw> {
-        name: Some(Ident::new(index_name)),
+        name: Some(Ident::new_unchecked(index_name)),
         on_name: RawItemName::Name(mz_sql::normalize::unresolve(view_name)),
         in_cluster: Some(RawClusterName::Resolved(cluster_id.to_string())),
         key_parts: Some(
             keys.iter()
                 .map(|i| match view_desc.get_unambiguous_name(*i) {
-                    Some(n) => Expr::Identifier(vec![Ident::new(n.to_string())]),
+                    Some(n) => Expr::Identifier(vec![Ident::new_unchecked(n.to_string())]),
```

Comment:
We should probably create a follow-up GH issue to try and remove these types of `new_unchecked` uses. It's not obvious to me that these are guaranteed to fit.

Reply:
Filed https://github.com/MaterializeInc/materialize/issues/23191 to track this. It's not obvious to me either, but with the `soft_assert!` in place, I feel much more confident that we'll atleast catch these issues earlier